src/instructlab/model/backends/backends.py

```diff
@@ -121,12 +154,18 @@ def determine_backend(model_path: pathlib.Path) -> str:
     Returns:
         str: The backend to use.
     """
-    # Check if the model is a GGUF file
+    # If the model is a safetensors model and OS is linux, it's a vLLM model
+    try:
+        is_safetensors = is_model_safetensors(model_path)
+    except Exception as e:
+        raise ValueError(
+            f"Failed to determine whether the model is a safetensors model: {e}"
+        ) from e
```

Comment:
```suggestion
if model_path.is_dir() and sys.platform == "linux":
    try:
        is_safetensors = is_model_safetensors(model_path)
    except Exception as e:
        raise ValueError(
            f"Failed to determine whether the model is a safetensors model: {e}"
        ) from e
```

By doing so you don't need to check for the `is_dir()` inside `is_model_safetensors`.